pkg/handlers/backup.go

```diff
@@ -147,6 +148,33 @@ func (h *Handler) GetBackup(w http.ResponseWriter, r *http.Request) {
 	}
 	getBackupResponse.BackupDetails = backups
 
+	thisBackup, err := snapshot.GetInstanceBackup(r.Context(), util.PodNamespace, mux.Vars(r)["snapshotName"])
```

Comment:
I believe this function already works for legacy backups because GetBackupName will return the metadata.Name if there is no backup name label

https://github.com/replicatedhq/kots/blob/3531b8d2f464a9696409a253d3112c99df4cc38c/pkg/kotsadmsnapshot/types/types.go#L205-L210
https://github.com/replicatedhq/kots/blob/3531b8d2f464a9696409a253d3112c99df4cc38c/pkg/kotsadmsnapshot/backup.go#L850